addons/website_slides/models/slide_slide.py

```diff
@@ -656,9 +659,13 @@ def _action_set_quiz_done(self):
                      slide.quiz_second_attempt_reward,
                      slide.quiz_third_attempt_reward,
                      slide.quiz_fourth_attempt_reward]
-            points += gains[user_membership_sudo.quiz_attempts_count - 1] if user_membership_sudo.quiz_attempts_count <= len(gains) else gains[-1]
+            points = gains[user_membership_sudo.quiz_attempts_count - 1] if user_membership_sudo.quiz_attempts_count <= len(gains) else gains[-1]
+            if points:
+                add_karma_result = self.env.user.sudo().add_karma(points, slide, _('Quiz completed'))
+            if not add_karma_result:
+                break
```

Comment:
as add_karma always returns true, unless it crashes on ensure_one or in the create, I don't see the point here.

Reply:
I agree.